### PR TITLE
API-Muncher Marisa Morris

### PR DESCRIPTION
**Pull request by marisamorris**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | Returning all of the data and then used nesting to step into to main hash further. |
| Describe your API Wrapper.  How did you decide on the methods you created? | My API wrapper contains two methods. A list_recipes method and a show recipe method. As I only needed to display a list of recipes and show an individual recipe, I didn't see the need for any additional. One change i could have made would have been to create a method that pulls the pieces of data i need so that each method can have a single responsibility.   |
| Describe an edge case or failure case test you wrote for your API Wrapper. | Will return [] for a broken request. If someone searches for nothing, an empty array is returned. |
| Explain how VCR aids in testing an API. | VCR is a tool that allows your tests to record HTTP interactions and replay them when necessary |
| What is the Heroku URL of your deployed application? | http://theguiltycarb.herokuapp.com/ |

### CODE REVIEW

File: test/lib/edamam_api_wrapper_test.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          recipes.must_be_instance_of Array
+          recipes.must_equal []
+        end
+      end
+
+  end
+
+  describe "show_recipe" do
+
+    it "Can show a requested recipe" do
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb]: What about a negative case?

File: test/controllers/home_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
+
+class HomepageControllerTest < ActionDispatch::IntegrationTest
```
Comment (CheezItMan) [inline test/controllers/home_controller_test.rb:3]: This should probably be removed.

File: test/controllers/recipes_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require 'test_helper'
+
+# class RecipesControllerTest < ActionDispatch::IntegrationTest
+describe RecipesController do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:4]: Again negative-case tests are needed!

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  <body>
+
+
+  <% unless controller_name == "home" %>
+    <nav>
+      <div class="grid-x">
+        <div class="large-2 medium-1 small-3 cell"></div>
+
+          <div class="large-8 medium-10 small-6 cell blah">
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:24]: `blah`?

File: lib/edamam_api_wrapper.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    end
+  end
+
+
+
+  def self.show_recipe(search)
+    # MUST ENCODE THE SEARCH PARAMETER IN THE URL
+    url = BASE_URL + "?r=" + (URI.encode(search))
+
+    data = HTTParty.get(url)
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb]: You need some error-detection here!  What if `data` or `data[0]` are `nil`?

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good number of commit and commit messages  |
| Comprehension questions	| Check, I would also emphasize that VCR keeps you from running too many api calls  |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | You've got a few too many `div` elements |
| Errors are reported to the user | Wrong show URLs crash the app, otherwise no errors reported |
| API Wrapper to handle the API requests | Check, but more defensive programming needed |
| Controller testing | Check, but no negative-case tests |
| Lib testing | Check, but no negative-case tests |
| Search Functionality | Check |
| List Functionality | Check |
| Show individual item functionality (link to original recipe opens in new tab) | Check |
| **Styling** |  |
| Foundation Styling for responsive layout | Your site simply shrinks the results as the browser gets smaller.  This results in an unpleasant experience on a mobile device.  Using grid-layout would be better.  Something to work on in JavaScript! |
| List View shows 10 items at a time/pagination | Nice work! |
| The app is styled to create an attractive user interface | The styling is pretty nice. |
| **API Features** | |
| The App attributes Edaman | Attribution is there, but no link |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to deployed app on Heroku | Check |
|  **Overall** | Overall well done.  You had some oversights in testing which expose problems in your app.  I left some notices in your code.  Overall not bad. |
